### PR TITLE
query node version and heartbeat in channel

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelImp.java [hunk 1]
```diff
@@ -222,13 +222,9 @@ public void onResponse(Response response) {
...
     public void asyncSendToPeer(Message out, String peerIpPort, ResponseCallback callback) {
         msgHandler.addSeq2CallBack(out.getSeq(), callback);
-        availablePeer.forEach(
-                (peer, ctx) -> {
-                    if (peer.equals(peerIpPort)) {
-                        ctx.writeAndFlush(out);
-                        logger.debug("send message to {} success ", peer);
-                    }
-                });
+        ChannelHandlerContext ctx = msgHandler.getAvailablePeer().get(peerIpPort);
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/sdk/channel/ChannelImp.java:225]: ctx maybe null

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelMsgHandler.java [hunk 2]
```diff
@@ -59,18 +83,14 @@ public void addSeq2CallBack(String seq, ResponseCallback callback) {
...
+        queryNodeVersion(ctx);
         for (MsgHandler handle : msgConnectHandlerList) {
             handle.onConnect(ctx);
         }
     }
 
     @Override
     public void onMessage(ChannelHandlerContext ctx, Message msg) {
-        // TODO: use msgHandlers to find special type to handle
         ResponseCallback callback = (ResponseCallback) seq2Callback.get(msg.getSeq());
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/sdk/channel/ChannelMsgHandler.java:94]: add getAndRemove(String seq) method for ResponseCallback  object

File: src/main/java/org/fisco/bcos/sdk/channel/model/EnumNodeVersion.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        public String getExt() {
+            return ext;
+        }
+
+        public void setExt(String ext) {
+            this.ext = ext;
+        }
+    }
+
+    public static Version getClassVersion(String version) throws ChannelPrococolExceiption {
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/sdk/channel/model/EnumNodeVersion.java:74]: Add support for sm node version number format, eg: 2.4.0 gm

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelHandlerContextHelper.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.util.AttributeKey;
+import org.fisco.bcos.sdk.channel.model.ChannelProtocol;
+import org.fisco.bcos.sdk.channel.model.EnumChannelProtocolVersion;
+import org.fisco.bcos.sdk.channel.model.EnumSocketChannelAttributeKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ChannelHandlerContextHelper {
```
Comment (ywy2090) [inline src/main/java/org/fisco/bcos/sdk/channel/ChannelHandlerContextHelper.java:12]: The class has a better name?

Comment (MaggieNgWu) [reply src/main/java/org/fisco/bcos/sdk/channel/ChannelHandlerContextHelper.java:12]: VersionNegotiateHelper ?

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelImp.java (hunk 1 above)
Comment (chaychen2005) [reply src/main/java/org/fisco/bcos/sdk/channel/ChannelImp.java:225]: check ctx before use

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelHandlerContextHelper.java (hunk 4 above)
Comment (chaychen2005) [reply src/main/java/org/fisco/bcos/sdk/channel/ChannelHandlerContextHelper.java:12]: ChannelHandlerContextHelper ->ChannelVersionNegotiation

File: src/main/java/org/fisco/bcos/sdk/channel/model/EnumNodeVersion.java (hunk 3 above)
Comment (chaychen2005) [reply src/main/java/org/fisco/bcos/sdk/channel/model/EnumNodeVersion.java:74]: if (version.endsWith("gm")) {
                version = version.substring(0, version.indexOf("gm"));
            }

File: src/main/java/org/fisco/bcos/sdk/channel/ChannelMsgHandler.java (hunk 2 above)
Comment (chaychen2005) [reply src/main/java/org/fisco/bcos/sdk/channel/ChannelMsgHandler.java:94]: Done

**Pull request merged**